subsys/net/lib/nrf_cloud/Kconfig

```diff
@@ -103,13 +103,35 @@ config NRF_CLOUD_MQTT_PAYLOAD_BUFFER_LEN
 	int "Size of the buffer for MQTT PUBLISH payload."
 	default 2048
 
+menuconfig NRF_CLOUD_FOTA
+	bool "Enable FOTA through nRF Cloud"
+	select FOTA_DOWNLOAD
+	select FOTA_DOWNLOAD_PROGRESS_EVT
+	select DFU_TARGET
+	select DOWNLOAD_CLIENT
+	select REBOOT
+	depends on SETTINGS
+	default y
+
+if NRF_CLOUD_FOTA
 config NRF_CLOUD_FOTA_PROGRESS_PCT_INCREMENT
 	int "Percentage increment at which FOTA download progress is reported"
-	depends on FOTA_DOWNLOAD_PROGRESS_EVT
 	range 0 100
 	default 10
 	help
 		0 disables progress report.
+
+config NRF_CLOUD_FOTA_DOWNLOAD_FRAGMENT_SIZE
+	int "Fragment size for FOTA downloads."
+	range 128 2000
+	default 1024
```

Comment:
Would it be safe to bump this up a bit to reduce the bytes sent back and forth? Depending on the HTTP header, perhaps something in the range 1700-1800 could work, but the risk is that the header grows over time, making the device un-FOTAable. In Azure FOTA this is worked around by adding the option to ship over the fragment size in the FOTA job document

Reply:
i'll test with some larger values.



> In Azure FOTA this is worked around by adding the option to ship over the fragment size in the FOTA job document

can you share the details of this?  i can see about getting that added to nrfcloud fota

Reply:
headers from modem update files are ~250 bytes... application update files are ~520 bytes 😅

Reply:
@jtguggedal  i've updated to 1700... which gives about 100 bytes of cushion.

however, something i don't quite understand:
if i use the [download](https://github.com/nrfconnect/sdk-nrf/tree/master/samples/nrf9160/download) sample and configure it for https and 2048 fragment size, it downloads just fine... and the http header size is ~730 bytes.

Reply:
> > In Azure FOTA this is worked around by adding the option to ship over the fragment size in the FOTA job document

> can you share the details of this? i can see about getting that added to nrfcloud fota

For Azure, we use the JSON schema that can be seen here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/include/net/azure_fota.html
The fragment size is then specified in the JSON document that described the FOTA sort of similar to how AWS Jobs work, just way simpler.

> @jtguggedal i've updated to 1700... which gives about 100 bytes of cushion.
> however, something i don't quite understand:
> if i use the download sample and configure it for https and 2048 fragment size, it downloads just fine... and the http header size is ~730 bytes.

Is this with TLS? I would expect it to work on non-TLS connection as that has 4k buffers in the modem, and fail with TLS with such a large HTTP header.

Reply:
@jtguggedal yes, TLS is enabled for the sample, i added a print for the http header size in http_parse in download_client:

```
*** Booting Zephyr OS build v2.4.0-ncs1-1715-g3366927a5498  ***
Download client sample started
Waiting for network.. OK
Downloading https://file-examples-com.github.io/uploads/2017/10/file_example_JPG_100kB.jpg
HTTP header len: 724
[   2% ] |=                                                 | (2048/102117 bytes) - HTTP header len: 726
[   4% ] |==                                                | (4096/102117 bytes) - HTTP header len: 726
[   6% ] |===                                               | (6144/102117 bytes) - HTTP header len: 726
...
```
I suppose inspecting this in wireshark would give me more insight.

Reply:
That's very confusing... Agree that Wireshark is probably needed to understand what's going on.